### ISSUE TITLE
Support DPI

### ISSUE THREAD

**Issue by ryantrem** [BabylonJS/BabylonNative#482]
Babylon.js has some support for scaling the display in different ways, based on the `adaptToDeviceRatio` `Engine` constructor parameter, `Engine.setHardwardScalingLevel`, and `window.devicePixelRatio`. I believe Babylon Native does not currently support any of this, and that currently we always specify view size in device pixels. I think that also means on mobile devices we are probably rendering at very high resolutions, perhaps higher than necessary with an associated resource cost. We should think about how to support this as it could have a huge impact on mobile performance.

**Comment by Drigax** [comment]
Sounds good. I'll start looking into this

**Comment by Drigax** [comment]
I added some rudimentary support for `NativeEngine.setHardwareScalingLevel` via this branch: https://github.com/Drigax/BabylonNative/tree/add-engine-hardware-scaling-level

I'll open a PR when I return

**Comment by Drigax** [comment]
Looking at MSDN:
https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio

```
The devicePixelRatio of Window interface returns the ratio of the resolution in physical pixels to the resolution in CSS pixels for the current display device. This value could also be interpreted as the ratio of pixel sizes: the size of one CSS pixel to the size of one physical pixel. In simpler terms, this tells the browser how many of the screen's actual pixels should be used to draw a single CSS pixel.
```

This gives me the impression that window.devicePixelRatio can be implemented native-side as another scalar applied to calculate the resolution of the BGFX window. I'll most likely implment this similarily to how I'm currently implementing `Engine.hardwareScalingLevel`

**Comment by ryantrem** [comment]
I think we can deduce `devicePixelRatio` on iOS/Android with:
iOS: [`UIScreen.mainScreen.nativeScale`](https://developer.apple.com/documentation/uikit/uiscreen/1617825-nativescale?language=objc)
Android: [`DisplayMetrics.density`](https://developer.android.com/reference/android/util/DisplayMetrics#density)
I'm not 100% sure this is right, so we'll need to test it out.

**Comment by Drigax** [comment]
https://github.com/BabylonJS/BabylonNative/pull/582
